cfme/base/__init__.py

```diff
@@ -64,7 +63,7 @@ def collect_logs(self):
     @property
     def zone(self):
         server_res = self.appliance.rest_api.collections.servers.find_by(id=self.sid)
-        server = server_res[0]
+        server = single(server_res)
```

Comment:
See comments about this `single` function below.

Reply:
Here is an instance where trying to use this new over complicated method will cause us problems. Right now, we're returning the first item in the list, and if its empty we'll hit an IndexError (fairly predictable).

Now this would raise one of two exception types that have to be explicitly imported, instead of handling built-in exception types.

If we're going to raise exception here, we should be raising the ValueError - or _maybe_ defining a unique exception in cfme/exceptions.py to raise.

More likely, we shouldn't be changing this behavior, and we should be in a test asserting that the servers REST collection find_by method only returns one item when passed a server ID.

Reply:
@mshriver 

I have seen many occurrences of `function()[0]` in our code-base. I don't think it is good practise to do this because there is no check whether we didn't get more then one item. So I came with `single` (maybe it is not the best name. Please suggest some better. I though about `only`, `only_one`) 

The `single` function is made to accept any iterable, including a generator, so it may look it is complicated, but I think this is payed off by it being quite generic. I planned to change other occurrences of `[0]` to use `single`.

I can certainly change it to raise ValueError or perhaps a subclass of ValueError. I don't care. I just wanted to see an exception when there are more values then expected rather than just make the code blindly continuing with whatever it found which causes various exceptions (randomly occurring, thus not so easy to debug) further in the code.